### PR TITLE
Feature/23 plane struct

### LINKED ISSUE #23: Add Plane struct and implement sphere-plane collision

### PR DESCRIPTION
**Pull request by isakvlarsson**
Closes #23

### CODE REVIEW

File: pphysx3d/src/shapes/bounding_volume.rs [hunk 1]
```diff
@@ -18,12 +18,12 @@ impl AABB {
     pub fn new(mins: Point3<f32>, maxs: Point3<f32>) -> AABB {
         AABB { mins, maxs }
     }
-      pub fn mins(&self) -> Point3<f32>{
```
Comment (coffecup25) [inline pphysx3d/src/shapes/bounding_volume.rs:21]: This man just wants to add more lines to his stats

File: pphysx3d/src/shapes/bounding_volume.rs [hunk 2]
```diff
@@ -18,12 +18,12 @@ impl AABB {
...
-      }
-      pub fn maxs(&self) -> Point3<f32>{
-            self.maxs
-      }
+    pub fn mins(&self) -> Point3<f32> {
+        self.mins
+    }
+    pub fn maxs(&self) -> Point3<f32> {
+        self.maxs
+    }
```
Comment (coffecup25) [inline pphysx3d/src/shapes/bounding_volume.rs:26]: This man just wants to add more lines to his stats

**Review by coffecup25 — APPROVED** [review]

File: pphysx3d/src/shapes/plane.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        return AABB::new(-maxs, maxs);
+    }
+    fn compute_bounding_sphere(&self, _pos: &Isometry3<f32>) -> BoundingSphere {
+        // Since a plane has infinite size, the BoundingSphere is a bit weird
+        let center: Point3<f32> = Point3::new(0.0, 0.0, 0.0);
+        return BoundingSphere::new(f32::MAX / 2.0, center);
+    }
+    fn as_sphere(&self) -> Option<&Sphere> {
+        return None;
+    }
```
Comment (M-Blomqvist) [inline pphysx3d/src/shapes/plane.rs:45]:
```suggestion
    fn as_sphere(&self) -> Result<&Sphere,()> {
        return Err(());
    }
    
    fn as_cube(&self) -> Result<&Cube,()> {
        return Err(());
    }
```

**Review by M-Blomqvist — CHANGES_REQUESTED** [review]
Probably still gonna break, should've pulld main

**Comment by isakvlarsson** [comment]
> Probably still gonna break, should've pulld main

What kind of changes do you want me to make? 🙂

**Review by M-Blomqvist — APPROVED** [review]

**Pull request merged**